cli/azd/pkg/infra/provisioning/bicep/bicep_provider.go

```diff
@@ -44,6 +47,14 @@ import (
 	"golang.org/x/exp/slices"
 )
 
+// cProjectNameDeploymentTagKey is the key name of the tag added to the deployment object that that holds the name of
+// of the azd project this deployment was created for (i.e. the name value from azure.yaml)
+const cProjectNameDeploymentTagKey = "azd-project-name"
```

Comment:
I worry if this might be unstable, and users might not know the consequences of renaming their project, which might be a reasonable thing to do. I could see this happening when the user is still playing around with a project, or maybe as part of a project refactor / rebrand.

I almost wonder if generating a unique environment ID might be the right thing to do, and that we should opening up to storing environment configuration inside a git repository that ends up allowing users on different machines to work on the same environment.